pkg/webui/console/containers/delete-entity-header-modal/index.js

```diff
@@ -95,8 +95,16 @@ const deletedErrorMessageMap = {
 }
 
 const DeleteEntityHeaderModal = props => {
-  const { entity, entityId, entityName, visible, setVisible, setError, additionalConditions } =
-    props
+  const {
+    entity,
+    entityId,
+    entityName,
+    visible,
+    setVisible,
+    setError,
+    additionalConditions,
+    additionalAction,
```

Comment:
Now that I look at this, `additionalConditions` and `additionalAction` seems a bit vague and hard to understand.

When it comes to gateway unclaiming, I think it's better to either add a prop called `unclaimCall` and then do it if the entity is a gateway or a prop called `supports claiming` and if true unclaim the gateway.

For the additional conditions I would call the prop `isPristine`. 

```suggestion
    isPristine: isPristineProp,
    supportsClaiming,
```